src/Security/Security.php

```diff
@@ -579,8 +579,13 @@ protected function getResponseController($title)
         $holderPage->URLSegment = 'Security';
         // Disable ID-based caching  of the log-in page by making it a random number
         $holderPage->ID = -1 * random_int(1, 10000000);
+        
+        if ($holderPage instanceof Controller) {
+            $controller = $holderPage;
```

Comment:
Should this go a bit higher in the method after the Injector call and move setting the title, url segment and ID into the condition when it's not an instance of a controller? Even though it might work it seems a bit hacky this way.

Reply:
I agree. Actually I believe that what would be nice is to set some kind of callback to generate the page. This way this code could be shifted into the CMS module altogether and avoid coupling entirely. But that requires coordinated changes between the framework and CMS module. It's not logical that the security class has knowledge of the page class.